src/Compiler/Compiler.cpp

```diff
@@ -378,6 +374,9 @@ bool Compiler::RegisterCommands(TclInterpreter* interp, bool batchMode) {
     std::string actualType = "VERILOG_2001";
     Design::Language language = Design::Language::VERILOG_2001;
     auto file = StringUtils::toLower(argv[1]);
+    if (file == "-work")  // if optional -work parameter goes first, the first
```

Comment:
very strange argument handling, would be good to improve this

Reply:
I moved that logic. now it's inside the loop and should be more readable